sdk/eventhub/event-hubs/src/tokenProvider.ts

```diff
@@ -2,33 +2,66 @@
 // Licensed under the MIT license.
 
 import { parseEventHubConnectionString } from "./util/connectionStringUtils";
-import { AccessToken } from "@azure/core-auth";
+import {
+  AccessToken,
+  AzureNamedKeyCredential,
+  AzureSASCredential,
+  NamedKeyCredential,
+  SASCredential
+} from "@azure/core-auth";
 import { Buffer } from "buffer";
 import isBuffer from "is-buffer";
 import jssha from "jssha";
 
+export interface TokenProvider {
+  /**
+   * Property used to distinguish TokenProvider from TokenCredential.
+   */
+  isTokenProvider: true;
+  getToken(audience: string): AccessToken;
+}
+
+/**
+ * Creates a token provider from the Event Hub connection string;
+ * @param connectionString - The Event Hub connection string
+ */
+export function createTokenProvider(connectionString: string): TokenProvider {
+  const parsed = parseEventHubConnectionString(connectionString);
+
+  if (parsed.sharedAccessSignature == null) {
+    return new NamedKeyTokenProvider(
+      new AzureNamedKeyCredential(parsed.sharedAccessKeyName!, parsed.sharedAccessKey!)
```

Comment:
TODO: I should be able to just pass an object like `{name: parsed.sharedAccessKeyName!, key: parsed.sharedAccessKey!}` instead of instantiating the credential. This would let me depend on `core-auth` purely for interfaces and not any runtime logic.